example/e2e/deploy-federated-deployment.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script has an end to end steps in deploying a federated deployment
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TEST_NS="test-namespace"
+KUBEFED_NS="kube-federation-system"
+
+# Prerequisite: a cluster scoped federation deployment with `cluster` and
+# `cluster2` joined as member clusters
+#
+# kubectl config use-context cluster1
+echo "Starting creating a ngnix application in federation cluster."
+
+echo "Creating a test namespace ${TEST_NS}."
+kubectl create ns ${TEST_NS}
+if [[ $? != 0 ]]; then
+    echo "Failed creating ${TEST_NS}."
+    exit 1
+fi 
+
+echo "Creating a ngnix deployment in namespace ${TEST_NS}."
+cat <<EOF | kubectl apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: ${TEST_NS}
+  name: nginx
+  labels:
+    app: nginx
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - image: nginx
+        name: nginx
+EOF
+
+if [[ $? != 0 ]]; then
+    echo "Failed creating a ngnix application."
+    exit 1
+fi 
+
+echo "Creating a ngnix service in namespace ${TEST_NS}."
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Service
+metadata:
+  namespace: ${TEST_NS}
+  name: nginx
+  labels:
+    app: nginx
+spec:
+  ports:
+  - nodePort: 30080
+    port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: nginx
+  type: NodePort
+EOF
+
+if [[ $? != 0 ]]; then
+    echo "Failed creating a ngnix service."
+    exit 1
+fi 
+
+echo "Federating resources in namespace ${TEST_NS} to member clusters."
+kubefedctl federate ns --federation-namespace ${KUBEFED_NS} --skip-api-resources 'pods,secrets,serviceaccount,replicaset' --contents ${TEST_NS}
```

Comment:
For readability I would write it as 
```
kubefedctl federate ns ${TEST_NS}  --contents --federation-namespace ${KUBEFED_NS} --skip-api-resources 'pods,secrets,serviceaccount,replicaset'
```

Reply:
Also the script defines `${KUBEFED_NS}` as a constant with value of `kube-federation-system` which is also the default. If the value is always going to be default, specifying the flag `--federation-namespace` is not necessary.

Reply:
done.